src/bot.py

```diff
@@ -16,50 +16,41 @@
 '''
 
 
-
-import os
-
-from telegram.ext import Updater, CommandHandler, MessageHandler, Filters, ConversationHandler, CallbackQueryHandler
+import time
 
 import Task
 import config
-from src.handler import CallbackQueryHandler as Cqh, CommandHandler as Ch, MessageHandler as Mh
+from src import _abc
+from src.backend.telegram import TelegramBackend
+
+
+BACKENDS: dict[str, type] = {'telegram': TelegramBackend}
+
+
+class Bot:
+    backends: dict[str, _abc.Backend]
+
+    def __init__(self, *args, **kwargs):
+        self.backends = {}
+        for name, backend in BACKENDS.items():
+            self.backends[name] = backend(bot=self, **kwargs.get(f"{name}_options", {}))
+        self.users = []
+
+    def run(self):
+        for backend in self.backends.values():
+            backend.run()
+
+        while True:
```

Comment:
This while block might be inefficient since the condition is always True. Also, doesn't `time.sleep()` cripple the program?
I'm currently using [APScheduler](https://pypi.org/project/APScheduler/) in a implicit way to schedule a task. python-telegram-bot provides it.

Reply:
> This while block might be inefficient since the condition is always True.

What do you mean? The while block lacking an exit condition is a problem but I am not sure if I understand what you mean by inefficient.

> Also, doesn't time.sleep() cripple the program?

The python-telegram-bot version you use(v13) uses threading, so running time.sleep in the main thread won't halt the execution of other threads. That'll be a concern if you upgrade to v20 which uses asyncio but changing the time.sleep to asyncio.sleep would easily fix that.

I thought about using the scheduler but the main bot's implementation must be backend-agnostic therefore we cannot use the telegram backend's scheduler and must initialize our own. And initializing a scheduler just for one task felt a bit wasteful to me but I can do that if you want.

Reply:
>  The while block lacking an exit condition is a problem but I am not sure if I understand what you mean by inefficient.

I guess "inefficiency" isn't the right word here, but people generally don't recommend using it without a proper cause. However, I read some threads on it, now I don't think it's a bad idea.

> I thought about using the scheduler but the main bot's implementation must be backend-agnostic therefore we cannot use the telegram backend's scheduler and must initialize our own. And initializing a scheduler just for one task felt a bit wasteful to me but I can do that if you want.

I see your point on *backend-agnostic* which is nice. Implementing our own scheduler can be helpful if we add new tasks with different intervals. Currently there are only one task but I hope there will be others in the future. See [to-do](https://github.com/furkansimsekli/hu-announcement-bot#to-do).